src/praisonai-agents/tests/unit/test_gateway_degraded_state.py

```
@@ -111,3 +113,81 @@ def test_owner_accepts_every_declared_vocabulary_value():
+    reg = DegradedCapabilityRegistry()
+    reg.mark(DegradedOwner("provider", "openai", "cold", "auth rejected", "fix"))
+    with pytest.raises(OwnerUnavailable):
+        reg.assert_owner_available("provider", "openai")
+    reg.clear("provider", "openai")
+    # Recovered: guard is a no-op again.
+    reg.assert_owner_available("provider", "openai")
+
+
+def test_module_level_guard_is_noop_for_none_registry():
+    # A gateway may run without a registry; the guard must not raise.
+    assert_owner_available(None, "provider", "openai") is None
```

Ruff (0.16.0)

[warning] Pointless comparison at end of function scope. Did you mean to return the expression result?

(B015)